frame/timestamp/src/lib.rs

```diff
@@ -189,7 +189,8 @@ pub mod pallet {
 		/// # </weight>
 		#[pallet::weight((
 			T::WeightInfo::set(),
-			DispatchClass::Mandatory
+			DispatchClass::Mandatory,
+			Pays::No,
```

Comment:
this makes sending a signed extrinsic with the call `set` free, which is not good, even if it will fails directly it can still fill some block for free